### PR TITLE
Fix bugs in client.go and log.go

### PR DESCRIPTION
**Pull request by AlverLyu**
1. Fix the out of range bug in client/client.go:SaveAccount when the length of private key is less than 32 bytes(the leading byte(s) is 0).
2. Check the pointer "Log" before used in Info, Warn, Error and Fatal.

### CODE REVIEW

File: common/log/log.go [hunk 1]
```diff
@@ -184,19 +184,27 @@ func Debug(a ...interface{}) {
 }
 
 func Info(a ...interface{}) {
-	Log.Info(fmt.Sprint(a...))
+	if Log != nil {
```
Comment (dreamfly281) [inline common/log/log.go]: which case will trigger this problem?

**Comment by AlverLyu** [comment]
It will be triggered when called before CreatePrintLog. Better to fix this to avoid misusing, e.g. in unit test.

**Comment by AlverLyu** [comment]
It will be triggered when called before CreatePrintLog. Better to fix this to avoid misusing, e.g. in unit test.在 2017年5月30日，00:00，Yanbo Li <notifications@github.com> 写道：@dreamfly281 commented on this pull request.

In common/log/log.go:
> @@ -184,19 +184,27 @@ func Debug(a ...interface{}) {
 }
 
 func Info(a ...interface{}) {
-	Log.Info(fmt.Sprint(a...))
+	if Log != nil {
which case will trigger this problem?—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/DNAProject/DNA","title":"DNAProject/DNA","subtitle":"GitHub repository","main_image_url":"<a href="https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png" class="">https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png</a>","avatar_image_url":"<a href="https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png" class="">https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png</a>","action":{"name":"Open in GitHub","url":"<a href="https://github.com/DNAProject/DNA" class="">https://github.com/DNAProject/DNA</a>"}},"updates":{"snippets":[{"icon":"PERSON","message":"@dreamfly281 commented on #257"}],"action":{"name":"View Pull Request","url":"<a href="https://github.com/DNAProject/DNA/pull/257#pullrequestreview-40800290" class="">https://github.com/DNAProject/DNA/pull/257#pullrequestreview-40800290</a>"}}}

**Comment by dreamfly281** [comment]
I don't think it is an elegant way fix this "issue",  most of modules need be initialization before calling it, ie DBFT, NET, Node and the Ledger,  if the test module call it without initialization the modules , it is the fault of the caller itself instead of the previous,  especially  for internal function calling, in team of API used by the external user, it is another case to add more condition  checking to avoid such kinds of problem, but for log modules, it is obviously a internal module.

**Comment by dreamfly281** [comment]
Looks good from my side, the only defect is the modification based on github webUI, it will lead to several useless commit and the description of the commit lack of detail.  The description of the log should be " what the issue it solved with what kinds of method" instead of simple word "bug fixing", hope it will not happened in the next time.

**Pull request merged**